tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -1374,6 +1377,635 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+			pvcs = append(pvcs, pvclaim)
+			scs = append(scs, storageclass)
+			pvclaim2, err := createPVC(client, namespace, nil, "", storageclass, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvcs = append(pvcs, pvclaim2)
+			pvcs2d = append(pvcs2d, []*v1.PersistentVolumeClaim{pvclaim})
+		}
+
+		defer func() {
+			ginkgo.By("Delete the SCs created in step 2")
+			for _, sc := range scs {
+				err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		volIds := []string{}
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i/2])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+			e2eVSphere.verifyDatastoreMatch(volumeID, []string{sharedvmfsURL})
+			volIds = append(volIds, volumeID)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			for _, pvc := range pvcs {
+				err := fpv.DeletePersistentVolumeClaim(client, pvc.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			for i := range pvcs {
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(pvs[i].Spec.CSI.VolumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		// Create a Pod to use this PVC, and verify volume has been attached
+		ginkgo.By("Creating 2 pods to attach 2 PVCs")
+		pods := createMultiplePods(ctx, client, pvcs2d, true)
+
+		defer func() {
+			ginkgo.By("Delete pods")
+			deletePodsAndWaitForVolsToDetach(ctx, client, pods, true)
+		}()
+
+		fsSizes := []int64{}
+		for _, pod := range pods {
+			originalSizeInMb, err := getFSSizeMb(f, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			fsSizes = append(fsSizes, originalSizeInMb)
+		}
+
+		ginkgo.By("Change volume allocation of pol1 and pol2 to EZT")
+		for i, policyId := range policyIds {
+			err = updateVmfsPolicyAlloctype(ctx, pc, eztAllocType, policyNames[i], policyId)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Apply the volume allocation changes and while it is still running, expand PVCs created to a" +
+			" large size such that the resize takes more than vpxd task timeout")
+		var wg sync.WaitGroup
+		wg.Add(len(volIds))
+		n := resource.MustParse(largeSize)
+		sizeInInt, b := n.AsInt64()
+		gomega.Expect(b).To(gomega.BeTrue())
+		// since we are creating 4 volumes reducing the size of each expansion by half to start with
```

Comment:
Can you please explain this logic? We are creating 2 volumes in this test right?

Reply:
we are creating 4, 2 with thin(1 for online, 1 for offline case) and 2 with lzt